### PR TITLE
add autogen.initiate_chats

### PR DESCRIPTION
**Pull request by qingyun-wu**
## Why are these changes needed?

Adding an `initiate_chats` API so that one can construct a sequence of chats with different senders (comparing to the `initiate_chats` method of a particular ConversableAgent). 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by afourney** [comment]
Looks like a really useful interface. 

I need to test it this week on my workflows, but I think it's a natural extension to the agent's initiate_chats mechanism from the week prior.

One natural question this PR raises though is: Do we still need initiate_chats in conversable_agent? It seems like this could completely replace it.

File: notebook/agentchats.ipynb
Comment (sonichi) [inline notebook/agentchats.ipynb:1]: Would you like to follow https://github.com/microsoft/autogen/blob/main/notebook/contributing.md
to make the notebook displayable on the website?
cc @jackgerrits

Comment (qingyun-wu) [reply notebook/agentchats.ipynb:1]: added

File: notebook/agentchats.ipynb [hunk 1]
```diff
@@ -0,0 +1,993 @@
...
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# Auto Generated Agent Chat: Multi Task Chats\n",
```
Comment (gagb) [inline notebook/agentchats.ipynb:12]: This title is confusing. How about:

"Solving a set of tasks with a sequence of chats"

File: notebook/agentchats.ipynb [hunk 2]
```diff
@@ -0,0 +1,993 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import autogen\n",
+    "\n",
+    "config_list = autogen.config_list_from_json(env_or_file=\"OAI_CONFIG_LIST\")\n",
+    "llm_config = {\"config_list\": config_list}\n",
```
Comment (gagb) [inline notebook/agentchats.ipynb:43]: Use the convention set by @jackgerrits

**Review by gagb — APPROVED** [review]

File: notebook/agentchats.ipynb (hunk 2 above)
Comment (jackgerrits) [reply notebook/agentchats.ipynb:43]: That's the right convention 
https://github.com/microsoft/autogen/blob/main/notebook/contributing.md#consistency-for-installation-and-llm-config

We considered using:
```python
llm_config = {
    "config_list": [{"model": "gpt-4", "api_key": os.environ["OPENAI_API_KEY"]}],
}
```
directly, but it is convenient to be able to use your predefined config directly in the notebooks, so I am thinking autogen.config_list_from_json is useful

**Comment by sonichi** [comment]
The test failed: https://github.com/microsoft/autogen/actions/runs/7881102985/job/21504193590?pr=1638

**Pull request merged**